src/test/scala/za/co/absa/hyperdrive/trigger/api/rest/services/WorkflowValidationServiceTest.scala

```diff
@@ -117,13 +139,60 @@ class WorkflowValidationServiceTest extends AsyncFlatSpec with Matchers with Moc
     )
   }
 
+  it should "fail if job template and job definition types are not equal" in {
```

Comment:
There should also be a test for `Job jar and main class cannot be empty in case of $jobType job type when template is empty`